libsolidity/ast/Types.cpp

```diff
@@ -2986,26 +2986,54 @@ TypePointer FunctionType::interfaceType(bool /*_inLibrary*/) const
 		return TypePointer();
 }
 
-bool FunctionType::canTakeArguments(TypePointers const& _argumentTypes, TypePointer const& _selfType) const
+bool FunctionType::canTakeArguments(
+	FuncCallArguments const& _arguments,
+	TypePointer const& _selfType
+) const
 {
 	solAssert(!bound() || _selfType, "");
 	if (bound() && !_selfType->isImplicitlyConvertibleTo(*selfType()))
 		return false;
 	TypePointers paramTypes = parameterTypes();
+	std::vector<std::string> const& paramNames = parameterNames();
+
 	if (takesArbitraryParameters())
 		return true;
-	else if (_argumentTypes.size() != paramTypes.size())
+	else if (_arguments.numArguments() != paramTypes.size())
 		return false;
-	else
+	else if (!_arguments.hasNamedArguments())
 		return equal(
-			_argumentTypes.cbegin(),
-			_argumentTypes.cend(),
+			_arguments.types->cbegin(),
+			_arguments.types->cend(),
 			paramTypes.cbegin(),
 			[](TypePointer const& argumentType, TypePointer const& parameterType)
 			{
 				return argumentType->isImplicitlyConvertibleTo(*parameterType);
 			}
 		);
+	else if (paramNames.size() != _arguments.numNames())
```

Comment:
Shouldn't that be done earlier?

Reply:
`numNames()` might very well be `0` if they aren't given at all. We only start caring about them after the check that there are named arguments at all.

Reply:
Ah sorry, I might have confused it with `numArguments`.